lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1645,6 +1645,10 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
   if (validateTBDIfNeeded(Invocation, MSF, *IRModule.getModule()))
     return true;
 
+  // Free up some resources now that we're about to generate code.
+  // At this point, the SILModule is no longer needed, so it can be freed.
+  SM.reset();
```

Comment:
I'm not sure this is valid, as `SM` has already been `std::move`'d into `generateIR`, and if I understand it correctly, `IRGenRequest` tries to free the SILModule:

https://github.com/apple/swift/blob/3c606c3236991f043b1ed73eff4d79d20252c1d8/lib/IRGen/IRGen.cpp#L1149-L1159

Is this not actually freeing it?

Reply:
Oh! Then this patch is probably outdated -- was upstreaming it from a fork I have. Probably means that this `reset()` call does nothing because the pointer has been moved. Will double-check before closing but thanks for the pointer!

Reply:
Yep, this is no longer needed at all -- you are 100% correct that `SM` has been moved into `generateIR` and is owned by that now. Closing this out.